homeassistant/auth/util.py

```diff
@@ -1,13 +0,0 @@
-"""Auth utils."""
-import binascii
-import os
-
-
-def generate_secret(entropy: int = 32) -> str:
```

Comment:
Can you double check this default is the same?

Reply:
We can start a crypto fight :D

> In particularly, secrets should be used in preference to the default pseudo-random number generator in the random module, which is designed for modelling and simulation, not security or cryptography.

What does that mean? All entropy is bad until the OS provides a good entropy and random generator. In an example, HassOS. We add hardware support and service they provide the hights and best entropy on OS random interfaces that are possible. However, whatever we do on our code, it's only so secure like the user/os can provide that for us.

Reply:
As the description of this method states, this method IS a backport of Python 3.6. However, I am not sure if I changed the default entropy.

Reply:
The default is 32 in secrets too, already checked before submitting this.
```
$ python3.6 -c 'import secrets;print(len(secrets.token_hex())/2)'
32.0
$ python3.7 -c 'import secrets;print(len(secrets.token_hex())/2)'
32.0
$ python3.8 -c 'import secrets;print(len(secrets.token_hex())/2)'
32.0
```